src/views/tutorial-view/index.tsx

```diff
@@ -33,64 +28,98 @@ import TutorialsSidebar, {
 import TabProvider from 'components/tabs/provider'
 import TutorialMeta from 'components/tutorial-meta'
 import VideoEmbed from 'components/video-embed'
-import InstruqtProvider from 'contexts/instruqt-lab'
-import { useOptInAnalyticsTracking } from 'hooks/use-opt-in-analytics-tracking'
 import { getLearnRedirectPath } from 'components/opt-in-out/helpers/get-learn-redirect-path'
 
 // Local imports
+import {
+	CollectionContext,
+	LayoutContentWrapperProps,
+	TutorialData,
+	TutorialSidebarSidecarProps,
+	TutorialViewProps,
+} from './types'
 import MDX_COMPONENTS from './utils/mdx-components'
 import { formatTutorialToMenuItem, generateCanonicalUrl } from './utils'
+import getVideoUrl from './utils/get-video-url'
+import { getCanonicalCollectionSlug } from './utils/get-canonical-collection-slug'
 import {
 	FeaturedInCollections,
 	NextPrevious,
 	getNextPrevious,
 } from './components'
-import getVideoUrl from './utils/get-video-url'
-import { getCanonicalCollectionSlug } from './utils/get-canonical-collection-slug'
 import s from './tutorial-view.module.css'
 
-export interface TutorialViewProps {
-	layoutProps: TutorialSidebarSidecarProps
-	product: LearnProductData
-	tutorial: TutorialData
-}
+/**
+ * The purpose of this wrapper component is to make it possible to invoke the
+ * `useMobileMenu` hook. It cannot be invoked in `TutorialView`. This is because
+ * it requires being invoked within a `MobileMenuProvider`.
+ *
+ * `MobileMenuProvider` is rendered by `CoreDevDotLayout`, which is indirectly
+ * rendered by `SidebarSidecarLayout`. This means the `useMobileMenu` hook can
+ * only be invoked here by a component rendered within `SidebarSidecarLayout`.
+ */
+const LayoutContentWrapper = ({
+	children,
+	collectionCtx,
+	product,
+	setCollectionViewSidebarSections,
+}: LayoutContentWrapperProps) => {
+	const { mobileMenuIsOpen } = useMobileMenu()
+	const hasLoadedData = useRef(false)
 
-export interface TutorialData
-	extends Pick<
-		ClientTutorial,
-		| 'name'
-		| 'slug'
-		| 'readTime'
-		| 'productsUsed'
-		| 'edition'
-		| 'handsOnLab'
-		| 'video'
-	> {
-	collectionCtx: CollectionContext
-	content: MDXRemoteSerializeResult
-	nextCollectionInSidebar?: ClientCollectionLite
-}
+	/**
+	 * Only need to load the data once, on the first open of the mobile menu
+	 */
+	useEffect(() => {
+		if (hasLoadedData.current === false && mobileMenuIsOpen) {
+			getCollectionViewSidebarSections(product, collectionCtx.current)
+				.then((result: CollectionCategorySidebarSection[]) => {
+					hasLoadedData.current = true
+					setCollectionViewSidebarSections(result)
+				})
+				.catch((error) => {
+					/**
+					 * TODO: revist how we're handling this error
+					 * https://app.asana.com/0/1202097197789424/1202599138117878/f
+					 */
+					const errorMessage = `[TutorialView] error calling \`getCollectionViewSidebarSections\`: ${JSON.stringify(
+						error
+					)}`
+					console.error(errorMessage)
+					throw new Error(errorMessage)
```

Comment:
💭: I don't think we want to re-throw here, otherwise we get an `Uncaught (in promise) Error`:
```
Uncaught (in promise) Error: [TutorialView] error calling `getCollectionViewSidebarSections`:
```

The `console.error` should be adequate!

Reply:
This is the discussion I'm trying to defer with the new ticket linked in the TODO. An error occurring provides information to the user, whereas an empty menu that they expect to be populated does not provide any information to them. I think we need to keep this for now until we have a wider discussion and firmer decision on what to do in the app when frontend errors occur.

Reply:
The error doesn't do anything other than log to the console, so it's only helpful if a user has the console open, in which case they'll see the message from `console.error`.

Here's what we see in the console now when I block the request:
![Screen Shot 2022-07-15 at 12 05 52 PM](https://user-images.githubusercontent.com/1289701/179273378-1a704314-db18-47b6-924f-4ec86d6d728e.png)

And the menu:
![Screen Shot 2022-07-15 at 12 07 00 PM](https://user-images.githubusercontent.com/1289701/179273538-4a61b87d-629c-408e-a66b-0e6332d5c079.png)

The empty menu is better than some unexpected/broken UI, and we still have the `console.error` which would provide insight if the console was open.

Reply:
When I do the same, I have an error that pops up in the viewport on top of everything else. This is what tells the user that something went wrong.

![CleanShot 2022-07-15 at 13 16 39](https://user-images.githubusercontent.com/43934258/179275194-ea78c2e8-98e8-40e6-85dd-91a0aa696ab0.gif)

Since the same message is logged to the console automatically from throwing the error, I can remove the redundant `console.error` for our sake since we read those.

Reply:
Won't our `ErrorBoundary` doesn't catch the error thrown?

Reply:
Ah, I think the `ErrorBoundary` doesn't catch in this case because we're using a `catch`? According to: https://reactjs.org/docs/error-boundaries.html#how-about-event-handlers

Reply:
Close! Error boundaries don't catch errors thrown in async code, so I don't think we would see the error boundary even without the catch.

Even so, I would argue the error boundary is a much worse experience for the user in this case as the entire page would be gone, whereas the page is still very much usable without these collection links being present in the menu.